src/metadata.rs

```diff
@@ -25,30 +25,165 @@ pub struct Metadata {
     imdb: Option<String>,
 }
 
+macro_rules! check_pattern {
+    ($pattern:ident, $name:ident, $title_start:ident, $title_end:ident, $get_capture:expr) => {
+        $pattern
+            .captures($name)
+            .map(|caps| {
+                if let Some(cap) = caps.get(0) {
+                    if $pattern.before_title() {
+                        $title_start = max($title_start, cap.end());
+                    } else {
+                        $title_end = min($title_end, cap.start());
+                    }
+                };
+                $get_capture(caps)
+            })
+            .flatten()
+    };
+    ($pattern:ident, $name:ident, $title_start:ident, $title_end:ident) => {
+        $pattern.captures($name).map(|caps| {
+            if let Some(cap) = caps.get(0) {
+                if $pattern.before_title() {
+                    $title_start = max($title_start, cap.end());
+                } else {
+                    $title_end = min($title_end, cap.start());
+                }
+            };
+            caps
+        })
+    };
+}
+
+macro_rules! match_to_string {
+    ($name:ident) => {
+        $name.and_then(|c| c.get(0)).map(|m| m.as_str().to_string())
+    };
+}
```

Comment:
These should probably be functions with inline markers rather than macroes.

Reply:
Switched them to functions - adding `#[inline]` markers actually causes a small (~3% on `names`, ~1% on `unicode`) but repeatable performance regression on my machine, I'm _guessing_ L1I$ churn but not sure